drafts/25-xxx-specifications.txt

```diff
@@ -0,0 +1,491 @@
+      6.3.2.4, or with a non-blank, non-zero character in column 6
+      of a fixed-form line as specified in Fortran 2023 6.3.3.3).
+
+      Example 1 (free-form):
+          call subroutine foo(1, 2, &
+      #ifdef USE_3
+                                    3, &
+      #else
+                                    666, &
+      #endif
+                                    .true.)
+
+      Example 2 (fixed-form):
+            call subroutine foo(1, 2,
+      #ifdef USE_3
+           1                       3,
+      #else
+           1                       666,
+      #endif
+           2                       .true.)
+
+
+
+2.1.2 Case sensitivity of identifiers
+-------------------------------------
+
+cs01. Directive names are case-sensitive and recognized in lower-case.
+
+cs02. Macro names and function-like macro argument names are
+      case-sensitive.
+
+cs03. Preprocessor-defined macro names are case sensitive.
+
+
+
+2.1.3 Significance of whitespace
+--------------------------------
+
+ws01. The whitespace characters blank and the tab character may appear
+      on directive lines.
+
+ws03. Whitespace characters are significant in determining token
+      boundaries in preprocessor directive lines.
+
+ws02. Whitespace characters are significant in determining token
+      boundaries in fixed-form and free-fort Fortran source fragments.
+
+ws03. Whitespace characters are significant in determining token
+      boundaries in fixed-form and free-fort Fortran comment lines.
+
+ws04. In fixed-form input, macro names are not recognized as such when
+      spaces are inserted into their invocations.
+
+
+2.2 Comments
+------------
+
+co01. Directive lines may contain C-style '/*' ... '*/' comments.
+
+co02. Directive lines shall not contain C style '//' comments.
+
+co02. '/*' ... '*/' comments on directive lines shall extend past a
+      new-line only if the line ends in '\' new-line, indicating a
+      continuation line. (Maybe controversial)
+
+co03. '/*' ... '*/' comments on directive lines are replaced by a
+      single space, as specified in C 2023 5.1.1.2 bullet 3.
+
+co04. Directive lines shall not contain Fortran '!' style comments.
+
+co05. Directive lines (by definition) cannot contain Fortran
+      fixed-form 'C' or '*' style comments.
+
+
+2.4 Token lexicon
+-----------------
+[gak: List the character sequences that are recognized as separate tokens
+on directives and in Fortran source fragments. This should include
+Fortran identifiers, Fortran numeric constants, Fortran character
+constants, the Fortran operators, and the C operators used to construct
+constant-integer-expression.]
+
+
+
+3 #-Directives
+==============
+
+di01. The '#define' macro directive
+
+di02. The '#define' function-like macro directive
+
+di03. The '#undef' directive
+
+di04. The '#include' directive
+
+di03. The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
+      '#elifndef', '#else', '#endif' directive collective
```

Comment:
The term `collective` already has an unrelated technical meaning in the Fortran spec (and HPC).

I suggest replacing it with the word "collection", or just dropping the word entirely:

```suggestion
di03. The '#if', '#ifdef', '#ifndef', '#elif', '#elifdef',
      '#elifndef', '#else', '#endif' directives
```